### DISCUSSION TITLE
Input filtering

### DISCUSSION THREAD

**Discussion by ghost** [laravel/framework#52398] [Q&A]
Phalcon php has something like
 filter:string 
which would call 
filter_var($x, FILTER_SANITIZE_STRING); 
in background.
Does Laravel offer something simillar?

**Comment by shaedrich** [comment]
It would be nice if form requests had a method, similar to `rules()` (where the [validation filters](https://www.php.net/manual/en/filter.filters.validate.php) would apply), that would define these transformations ([sanitize filters](https://www.php.net/manual/en/filter.filters.sanitize.php)) which would be applied when `$request->validated()` is called.

**Reply by ghost** [reply]
Not only form requests.

**Reply by shaedrich** [reply]
Well, you can always call `filter_var()` directly

Or how would you want this to be implemented?

**Reply by ghost** [reply]
I could yes. But FILTER_SANITIZE_STRING is deprecated as of php 8.1. It was very handy until now. FILTER_UNSAFE_RAW is an alternative but it does not behave in the same way.

**Reply by ghost** [reply]
https://phalcon-php-framework-documentation.readthedocs.io/en/latest/reference/filter.html

**Reply by shaedrich** [reply]
Ah, okay. I thought, you were asking about the method in general and the constant argument was just an example. Sorry for the misunderstanding!

This is what I could find: https://stackoverflow.com/questions/69207368/constant-filter-sanitize-string-is-deprecated
Apparently, it suggests using either `FILTER_SANITIZE_FULL_SPECIAL_CHARS` or `htmlspecialchars()`

However, maybe, a Laravel implementation should be more configurable, meaning, you would be able to pass all kinds of sub filters to the overall string filter 🤔 Perhaps, there's already a package for that that someone might know about

**Reply by ghost** [reply]
FILTER_SANITIZE_FULL_SPECIAL_CHARS behaves weird. I used that first than switched to FILTER_UNSAFE_RAW.

**Reply by shaedrich** [reply]
> However, maybe, a Laravel implementation should be more configurable, meaning, you would be able to pass all kinds of sub filters to the overall string filter 🤔 Perhaps, there's already a package for that that someone might know about

This could be powered by the `Str` class, which already has some methods that could be used for input sanitizing:
* `squish()`/`trim()`
* `ascii()`
* `remove()`
* `slug()`
* `take()`
* `pipe()`
* `replace()`
* `scan()`
